keptn-operator/controllers/keptnservicecontroller/keptnservice_controller.go

```diff
@@ -240,44 +231,36 @@ func (r *KeptnServiceReconciler) createService(ctx context.Context, service stri
 	return err
 }
 
-func (r *KeptnServiceReconciler) checkIfServiceExists(ctx context.Context, req ctrl.Request, project string, service string) bool {
+func (r *KeptnServiceReconciler) checkIfServiceExists(project string, service string) (bool, error) {
 
-	projectsHandler := apiutils.NewAuthenticatedProjectHandler(r.KeptnAPI, utils.GetKeptnToken(ctx, r.Client, r.ReqLogger, req.Namespace), "x-token", nil, r.KeptnAPIScheme)
-	servicesHandler := apiutils.NewAuthenticatedServiceHandler(r.KeptnAPI, utils.GetKeptnToken(ctx, r.Client, r.ReqLogger, req.Namespace), "x-token", nil, r.KeptnAPIScheme)
+	projectsHandler := apiutils.NewAuthenticatedProjectHandler(r.KeptnInstance.Spec.APIUrl, r.KeptnAPIToken, r.KeptnInstance.Status.AuthHeader, nil, r.KeptnInstance.Status.Scheme)
```

Comment:
eventually things like the `projectsHandler` and `serviceHandler` should be injected into the `KeptnServiceReconciler`, instead of initialising them here